lib/seed/reservation_generator.rb

```diff
@@ -0,0 +1,35 @@
+module ReservationGenerator
+  extend ReservationGeneratorHelper
+
+  RES_TYPES = Reservation.statuses.keys + %w(future overdue returned_overdue) -
+              %w(reserved)
+
+  # Generate one of each reservation type with fixed time differences and
+  # one with random time differences
+  def self.generate_all_types
+    RES_TYPES.each do |t|
+      attempt_res_gen(t)
+      attempt_res_gen(t, true)
+    end
+  end
+
+  # Generate random reservation
+  def self.generate_random
+    attempt_res_gen(RES_TYPES.sample, true)
+  end
+
+  def self.attempt_res_gen(type, random = false)
+    50.times do
```

Comment:
I think that this is probably the best solution given the way random reservations are generated.

And yes, that's why we attempt reservation creation